Vienna/Sources/Main window/FolderTreeCellView.m

```diff
@@ -0,0 +1,234 @@
+// status to the text field however. Since the whole frame of the cell view
+// responds to events, the text field is activated needlessly when the user
+// clicks anywhere within the frame. Disabling first responder avoids that.
+- (BOOL)acceptsFirstResponder
+{
+    return NO;
+}
+
+// In macOS 12 this property returns a value based on the text field's first
+// baseline. That value is used by the disclosure triangle. This behavior is
+// not working correctly when the row height is changed. Returning 0 ensures
+// that the disclosure triangle is vertically centered.
+- (CGFloat)firstBaselineOffsetFromTop
+{
+    return 0.0;
+}
+
+// MARK: Accessors
+
+- (void)setSize:(VNAFolderListCellViewSize)size
+{
+    _size = size;
+    [self updateViewForSize];
+}
+
+- (void)setEmphasized:(BOOL)emphasized
+{
+    _emphasized = emphasized;
+    [self updateViewForFontTraits];
+}
+
+- (void)setInactive:(BOOL)inactive
+{
+    _inactive = inactive;
+    if (_inactive) {
+        self.textField.textColor = NSColor.tertiaryLabelColor;
+    } else {
+        self.textField.textColor = NSColor.labelColor;
+    }
+}
+
+- (void)setInProgress:(BOOL)inProgress
+{
+    _inProgress = inProgress;
+    if (_inProgress) {
+        self.errorImageView.hidden = YES;
+        self.progressIndicator.hidden = NO;
+        [self.progressIndicator startAnimation:nil];
+    } else {
+        [self.progressIndicator stopAnimation:nil];
+        self.progressIndicator.hidden = YES;
+    }
+}
+
+- (void)setShowError:(BOOL)showError
+{
+    _showError = showError;
+    if (_showError) {
+        self.errorImageView.hidden = NO;
+    } else {
+        self.errorImageView.hidden = YES;
+    }
+}
+
+- (void)setUnreadCount:(NSInteger)unreadCount
+{
+    _unreadCount = MAX(unreadCount, 0);
+    [self updateUnreadCountWithAnimation:NO];
+}
+
+- (void)setCanShowUnreadCount:(BOOL)canShowUnreadCount
+{
+    _canShowUnreadCount = canShowUnreadCount;
+    BOOL animate = NSAnimationContext.currentContext.allowsImplicitAnimation;
+    [self updateUnreadCountWithAnimation:animate];
+}
+
+// MARK: Utility
+
+- (NSFont *)fontForTextStyle:(NSFontTextStyle)textStyle NS_AVAILABLE_MAC(11)
+{
+    NSFontDescriptor *descriptor;
+    descriptor = [NSFontDescriptor preferredFontDescriptorForTextStyle:textStyle
+                                                               options:@{}];
+    NSFont *font = [NSFont fontWithDescriptor:descriptor
+                                         size:descriptor.pointSize];
+    return font;
+}
+
+- (NSImageSymbolConfiguration *)symbolForTextStyle:(NSFontTextStyle)textStyle
+    NS_AVAILABLE_MAC(11)
+{
+    NSImageSymbolConfiguration *config;
+    config = [NSImageSymbolConfiguration configurationWithTextStyle:textStyle];
+    return config;
+}
+
+- (void)updateViewForSize
+{
+    switch (self.size) {
+    case VNAFolderListCellViewSizeTiny:
+        self.textFieldLeadingSpace.constant = 22.0;
```

Comment:
@Eitot I feel like that magic numbers such as these should be put in a `#DEFINE` at the start of the file. Or perhaps defined somewhere as a `const`.

What would you prefer?

Reply:
I will do that. I have to go over the commits and clean up the commit history anyway.